pkg/apiserver/authentication/options/authenticate_options.go

```diff
@@ -32,23 +33,21 @@ type AuthenticationOptions struct {
 	// maximum retries when authenticate failed
 	MaxAuthenticateRetries int `json:"maxAuthenticateRetries" yaml:"maxAuthenticateRetries"`
 
-	// token validation duration, will refresh token expiration for each user request
-	// 0 means never expire
-	TokenExpiration time.Duration `json:"tokenExpiration" yaml:"tokenExpiration"`
-
 	// allow multiple users login at the same time
 	MultipleLogin bool `json:"multipleLogin" yaml:"multipleLogin"`
 
 	// secret to signed jwt token
-	JwtSecret string `json:"jwtSecret" yaml:"jwtSecret"`
+	JwtSecret string `json:"-" yaml:"jwtSecret"`
+
+	OAuthOptions *oauth.Options `json:"oauthOptions" yaml:"oauthOptions"`
```

Comment:
I don't think all configurations need to be configured from the command line. OAuth options is a complex struct.

Reply:
command line arguments are useful for developing and debugging

Reply:
Configuration is too complex to be configured from the command line.

Reply:
If we can read configuration from file, we do it through command line arguments too, please add to command line options, especially it's very convenient for debbuging.

Reply:
Or you can expose only necessary command line options for debugging

Reply:
we can config the oauth client and idetify provider in kubesphere.yaml like:
```
authentication:
  authenticateRateLimiterMaxTries: 5
  authenticationRateLimiterDuration: 30m0s
  maxAuthenticateRetries: 6
  multipleLogin: false
  jwtSecret: "xxxxx"
  oauthOptions:
    identityProviders:
      - name: github
        mappingMethod: auto
        loginRedirect: true
        type: Github
        github:
          clientID: de6ff7bed0304e487b6e
          clientSecret: xxxxxx-xxxxx-xxxxx
          endpoint:
            authURL: https://github.com/login/oauth/authorize
            tokenURL: https://github.com/login/oauth/token
          redirectURL: https://ks-console.kubesphere-system.svc/oauth/callbak/github
          scopes:
            - user
```
there is no necessary command line options for debugging, set the configuration from file from command line is also unnecessary, why not use the kubesphere.yaml,  you can just defined the oauth options. Otherwise, what is kubesphere.yaml for.